File: Daily Challanges/2026-06-June/2026-06-30.js
function duplicateCharacterCount(str1, str2) {
  const chars = new Set(str1);
  let count = 0;
  for (const c of str2) {
    if (chars.has(c)) count++;
  }
  return count;
}

console.log(duplicateCharacterCount("aloha", "hei"));
console.log(duplicateCharacterCount("jambo", "bonjour"));

File: Daily Challanges/2026-06-June/2026-06-30.py
def duplicate_character_count(str1, str2):
    str1_set = set(str1)
    dup_count = 0
    for letter in str2:
        if letter in str1_set:
            dup_count += 1
    return dup_count

print(duplicate_character_count("aloha", "hei"))
print(duplicate_character_count("jambo", "bonjour"))
print(duplicate_character_count("hello", "hola"))